diff --git a/dist/index.js b/dist/index.js
index 7636e7a63c38325c3c2a3881554d9a227676fcf9..434804c16ca7ef87b71243a20cd607b0ff009168 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -7271,6 +7271,8 @@ function isMetaMask(ethereum) {
     return false;
   if (ethereum.isImToken)
     return false;
+  if (ethereum.isKrakenWallet)
+    return false;
   if (ethereum.isKuCoinWallet)
     return false;
   if (ethereum.isMathWallet)
diff --git a/dist/types/utils.d.ts b/dist/types/utils.d.ts
index 0168b06ae3b31ea20e1f41bec7c6e4ea3fcc4c51..44ec125aee1648470fe27510f83fa5baab6797e2 100644
--- a/dist/types/utils.d.ts
+++ b/dist/types/utils.d.ts
@@ -9,7 +9,7 @@ export type Mutable<type extends object> = {
 };
 /** Strict version of built-in Omit type */
 export type Omit<type, keys extends keyof type> = Pick<type, Exclude<keyof type, keys>>;
-export type WalletProviderFlags = 'isApexWallet' | 'isAvalanche' | 'isBackpack' | 'isBifrost' | 'isBitKeep' | 'isBitski' | 'isBlockWallet' | 'isBraveWallet' | 'isCoinbaseWallet' | 'isDawn' | 'isEnkrypt' | 'isExodus' | 'isFrame' | 'isFrontier' | 'isGamestop' | 'isHyperPay' | 'isImToken' | 'isKuCoinWallet' | 'isMathWallet' | 'isMetaMask' | 'isNestWallet' | 'isOkxWallet' | 'isOKExWallet' | 'isOneInchAndroidWallet' | 'isOneInchIOSWallet' | 'isOpera' | 'isPhantom' | 'isPortal' | 'isRabby' | 'isRainbow' | 'isStatus' | 'isTally' | 'isTokenPocket' | 'isTokenary' | 'isTrust' | 'isTrustWallet' | 'isXDEFI' | 'isZerion' | 'isTalisman' | 'isZeal' | 'isCoin98' | 'isMEWwallet' | 'isSafeheron' | 'isSafePal' | '__seif';
+export type WalletProviderFlags = 'isApexWallet' | 'isAvalanche' | 'isBackpack' | 'isBifrost' | 'isBitKeep' | 'isBitski' | 'isBlockWallet' | 'isBraveWallet' | 'isCoinbaseWallet' | 'isDawn' | 'isEnkrypt' | 'isExodus' | 'isFrame' | 'isFrontier' | 'isGamestop' | 'isHyperPay' | 'isImToken' | 'isKrakenWallet' | 'isKuCoinWallet' | 'isMathWallet' | 'isMetaMask' | 'isNestWallet' | 'isOkxWallet' | 'isOKExWallet' | 'isOneInchAndroidWallet' | 'isOneInchIOSWallet' | 'isOpera' | 'isPhantom' | 'isPortal' | 'isRabby' | 'isRainbow' | 'isStatus' | 'isTally' | 'isTokenPocket' | 'isTokenary' | 'isTrust' | 'isTrustWallet' | 'isXDEFI' | 'isZerion' | 'isTalisman' | 'isZeal' | 'isCoin98' | 'isMEWwallet' | 'isSafeheron' | 'isSafePal' | '__seif';
 export type WalletProvider = Evaluate<EIP1193Provider & {
     [key in WalletProviderFlags]?: true | undefined;
 } & {
diff --git a/dist/wallets/walletConnectors/chunk-NKVTQ5YX.js b/dist/wallets/walletConnectors/chunk-NKVTQ5YX.js
index 95c9f7dc645cd0584097986a2cfaeba68a4471b8..cfba4aa80bdba46e671a199e8c5a1244cc9f325d 100644
--- a/dist/wallets/walletConnectors/chunk-NKVTQ5YX.js
+++ b/dist/wallets/walletConnectors/chunk-NKVTQ5YX.js
@@ -2,29 +2,24 @@
 import {
   getWalletConnectConnector
 } from "./chunk-75A5A7MH.js";
+import {
+  getInjectedConnector,
+  hasInjectedProvider
+} from "./chunk-MBBGZGXF.js";
+import {
+  isMobile
+} from "./chunk-RETKWSKD.js";
 
 // src/wallets/walletConnectors/krakenWallet/krakenWallet.ts
 var krakenWallet = ({
-  projectId,
-  walletConnectParameters
-}) => ({
-  id: "kraken",
-  name: "Kraken Wallet",
-  iconUrl: async () => (await import("./krakenWallet-DAOXUKWO.js")).default,
-  iconBackground: "#FFD8EA",
-  downloadUrls: {
-    ios: "https://apps.apple.com/us/app/kraken-wallet/id1626327149",
-    mobile: "https://kraken.com/wallet",
-    qrCode: "https://kraken.com/wallet"
-  },
-  mobile: {
-    getUri: (uri) => {
-      return `krakenwallet://wc?uri=${encodeURIComponent(uri)}`;
-    }
-  },
-  qrCode: {
-    getUri: (uri) => uri,
-    instructions: {
+                      projectId,
+                      walletConnectParameters
+                    }) => {
+  const isKrakenWalletInjected = isMobile() && hasInjectedProvider({ flag: 'isKrakenWallet'})
+  const shouldUseWalletConnect = !isKrakenWalletInjected;
+  let qrConnector = void 0;
+  if (shouldUseWalletConnect) {
+    qrConnector = {
       learnMoreUrl: "https://kraken.com/wallet",
       steps: [
         {
@@ -44,12 +39,34 @@ var krakenWallet = ({
         }
       ]
     }
-  },
-  createConnector: getWalletConnectConnector({
-    projectId,
-    walletConnectParameters
-  })
-});
+  }
+
+  const getUriMobile = (uri) => {
+    return `krakenwallet://wc?uri=${encodeURIComponent(uri)}`;  };
+
+  const mobileConnector = {
+    getUri: shouldUseWalletConnect ? getUriMobile : void 0
+  };
+
+  return {
+    id: "kraken",
+    name: "Kraken Wallet",
+    iconUrl: async () => (await import("./krakenWallet-DAOXUKWO.js")).default,
+    iconBackground: "#FFD8EA",
+    downloadUrls: {
+      ios: "https://apps.apple.com/us/app/kraken-wallet/id1626327149",
+      mobile: "https://kraken.com/wallet",
+      qrCode: "https://kraken.com/wallet"
+    },
+    mobile: mobileConnector,
+    qrCode: qrConnector,
+    createConnector: shouldUseWalletConnect ? getWalletConnectConnector({
+      projectId,
+      walletConnectParameters
+    }) : getInjectedConnector({ flag: 'isKrakenWallet' })
+  }
+
+};
 
 export {
   krakenWallet
diff --git a/dist/wallets/walletConnectors/krakenWallet/krakenWallet.js b/dist/wallets/walletConnectors/krakenWallet/krakenWallet.js
index ff8e11190e78cda8b9f0acc7212206366f448fcc..b97cd666c09971aee63ba131890002dd0a42a17c 100644
--- a/dist/wallets/walletConnectors/krakenWallet/krakenWallet.js
+++ b/dist/wallets/walletConnectors/krakenWallet/krakenWallet.js
@@ -2,6 +2,7 @@
 import {
   krakenWallet
 } from "../chunk-NKVTQ5YX.js";
+import "../chunk-RETKWSKD.js";
 import "../chunk-75A5A7MH.js";
 export {
   krakenWallet
